src/test/java/bio/terra/service/duos/DuosClientTest.java

```diff
@@ -0,0 +1,112 @@
+package bio.terra.service.duos;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+import bio.terra.app.configuration.DuosConfiguration;
+import bio.terra.common.category.Unit;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.duos.exception.DuosDatasetBadRequestException;
+import bio.terra.service.duos.exception.DuosDatasetNotFoundException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+@RunWith(SpringRunner.class)
+@ActiveProfiles({"google", "unittest"})
+@Category(Unit.class)
+public class DuosClientTest {
+
+  @MockBean private DuosConfiguration duosConfiguration;
+  @Mock private RestTemplate restTemplate;
+
+  private static final String BASE_PATH = "https://consent.dsde-dev.broadinstitute.org";
+  private static final String DUOS_ID = "DUOS-123456";
+  private static final Integer DUOS_DATASET_ID = 1;
+
+  private DuosClient duosClient;
+  private AuthenticatedUserRequest userReq;
+
+  @Before
+  public void setup() {
+    duosClient = new DuosClient(duosConfiguration, restTemplate);
+
+    when(duosConfiguration.getBasePath()).thenReturn(BASE_PATH);
+
+    userReq =
```

Comment:
FWIW, we often have a static final with the default user to use.  I actually wonder if we should have a fixture that just has a common static user so we don't need to keep redefining it

Reply:
I had this thought too!

Reply:
At some point you should update to the latest version of terra-common-lib, which deprecates `AuthenticatedUserRequest` and replaces it with a wrapper around a bearer token string. It will be a big change since this object is used everywhere but it helps in cases like this; only the bearer token is used so it's unnecessary to set email and subject ID.

Reply:
Until we migrate to latest `terra-common-lib` I introduced static method `AuthenticationFixtures.randomUserRequest()` for reuse.

Reply:
Unless you need multiple different users for a single test, you could also create a static `user` object instead of making a new one for each test.